opensearch-operator/pkg/builders/cluster.go

```diff
@@ -147,7 +147,7 @@ func NewSTSForNodePool(cr *opsterv1.OpenSearchCluster, node opsterv1.NodePool, v
 							},
 
 							Name:  cr.Name,
-							Image: "opensearchproject/opensearch:1.0.0",
+							Image: DockerImageForCluster(cr),
```

Comment:
I've got an implementation for this in https://github.com/Opster/opensearch-k8s-operator/pull/60/commits/89ee8a94805e02cf23ac57e4d53f248edf2d3a7a so we can probably skip this in favour of that.

Reply:
I'd keep my change in there as it is only minimal (was more of a dummy) with the understanding that you overwrite it with the real implementation in your PR for the upgrade reconciler.

Reply:
Sure, the only reason I mentioned to leave it would be to avoid dealing with the merge conflict 🤣